pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go

```diff
@@ -183,43 +184,53 @@ func createRawIngress(scheme *runtime.Scheme, isvc *v1beta1api.InferenceService,
 			})
 			return nil, nil
 		}
-		host, err := generateIngressHost(ingressConfig, isvc, string(constants.Explainer), true)
+		explainerName := constants.ExplainerServiceName(isvc.Name)
+		err := client.Get(context.TODO(), types.NamespacedName{Name: constants.DefaultExplainerServiceName(isvc.Name), Namespace: isvc.Namespace}, existing)
+		if err == nil {
+			explainerName = constants.DefaultExplainerServiceName(isvc.Name)
+			predictorName = constants.DefaultPredictorServiceName(isvc.Name)
+		}
+		host, err := generateIngressHost(ingressConfig, isvc, string(constants.Explainer), true, explainerName)
 		if err != nil {
 			return nil, fmt.Errorf("failed creating top level explainer ingress host: %v", err)
 		}
-		explainerHost, err := generateIngressHost(ingressConfig, isvc, string(constants.Explainer), false)
+		explainerHost, err := generateIngressHost(ingressConfig, isvc, string(constants.Explainer), false, explainerName)
 		if err != nil {
 			return nil, fmt.Errorf("failed creating explainer ingress host: %v", err)
 		}
 		// :predict routes to the predictor when there is only predictor and explainer
 		if len(isvc.Spec.Predictor.Containers) != 0 && len(isvc.Spec.Predictor.Containers[0].Ports) != 0 {
-			rules = append(rules, generateRule(host, constants.DefaultPredictorServiceName(isvc.Name), "/", isvc.Spec.Predictor.Containers[0].Ports[0].ContainerPort))
+			rules = append(rules, generateRule(host, predictorName, "/", isvc.Spec.Predictor.Containers[0].Ports[0].ContainerPort))
 		} else {
-			rules = append(rules, generateRule(host, constants.DefaultPredictorServiceName(isvc.Name), "/", constants.CommonDefaultHttpPort))
+			rules = append(rules, generateRule(host, predictorName, "/", constants.CommonDefaultHttpPort))
 		}
-		rules = append(rules, generateRule(explainerHost, constants.DefaultExplainerServiceName(isvc.Name), "/", constants.CommonDefaultHttpPort))
+		rules = append(rules, generateRule(explainerHost, explainerName, "/", constants.CommonDefaultHttpPort))
 	} else {
-		host, err := generateIngressHost(ingressConfig, isvc, string(constants.Predictor), true)
+		err := client.Get(context.TODO(), types.NamespacedName{Name: constants.DefaultPredictorServiceName(isvc.Name), Namespace: isvc.Namespace}, existing)
+		if err == nil {
+			predictorName = constants.DefaultPredictorServiceName(isvc.Name)
+		}
```

Comment:
Is this needed ? seems `predictorName` can be used directly

Reply:
This is required to check the predictor service already exists with the default suffix.

Reply:
@yuzisun updated!